File: Leet 1791/solution.py
class Solution:
    def findCenter(self, edges: list[list[int]]) -> int:
        if len(edges) < 2:
            return -1
        return (set(edges[0]) & set(edges[1])).pop()
    

def main() -> None:
    print("1791. Find Center of Star Graph")
    sol = Solution()

    case_1: int = sol.findCenter([[1,2], [2,3], [4,2]])
    print(f"case 1 should equal 2? {case_1}")

    case_2: int = sol.findCenter([[1,2],[5,1],[1,3],[1,4]])
    print(f"case 2 should equal 1? {case_2}")
    

if __name__ == "__main__":
    main()

File: Leet 1791/solution.cs
namespace Leet1791;

using System;
using System.Collections.Generic;
using System.Linq;

class Solution
{
    public int FindCenter(int[][] edges) 
    {
        if (edges.Length < 2)
        {
            return -1;
        }
        
        HashSet<int> edgeFirst = new(edges[0]);
        edgeFirst.IntersectWith(edges[1]);
        
        if (edgeFirst.Any())
        {
            return edgeFirst.First();
        }
        
        return -1;
    }
}


class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("1791. Find Center of Star Graph");
        Solution solution = new();
        int[][] edges = new int[][]
        {
            [1, 2],
            [2, 3],
            [4, 2]
        };
        int center = solution.FindCenter(edges);
        Console.WriteLine($"Center of the star graph should equal 2? {center}");
    }
}